tests/rbd/rbd_utils.py

```diff
@@ -179,6 +179,45 @@ def snap_create(self, pool_name, image_name, snap_name):
         cmd = f"rbd snap create {pool_name}/{image_name}@{snap_name}"
         self.exec_cmd(cmd=cmd)
 
+    def snap_ls(self, pool_name, image_name, snap_name=None):
+        """
+        Lists the snapshots present for an image and returns the given snap_name if present
+        Args:
+            pool_name: name of the pool containing snapshots
+            image_name: name of the image whose snapshots are to be listed
+            snap_name: if specified will return the snapshot if its present
+
+        Returns:
+            the snapshots if present for given details else None
+        """
+
+        cmd = f"rbd snap ls {pool_name}/{image_name} --format json"
+        out = self.exec_cmd(cmd=cmd, output=True)
+        if not snap_name:
+            return out
+        else:
+            snaps = json.loads(out)
+            for snap in snaps:
+                return snap # if the snap matches the snap_name given
```

Comment:
Can we modify method to return actual snap list if users wishes to get the actual list ?

Reply:
If user does not specify a specific snapshot name, then we will return the entire list, else we will return the exact snap specified.